From e3b9931df65985699e492118a3f079f9a7fbcd73 Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Mon, 21 May 2018 10:48:03 +0800
Subject: [PATCH 3856/5242] MLK-18355-7: hdmi_rx: Enable HDMI RX CEC clock

commit  bb4ad218d6e3f45c460133495145dc057f291dee from
https://source.codeaurora.org/external/imx/linux-imx.git

Enable HDMI RX CEC clock when hdmi rx driver initialized.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/hdmi/mxc-hdmi-hw.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/media/platform/imx8/hdmi/mxc-hdmi-hw.c b/drivers/media/platform/imx8/hdmi/mxc-hdmi-hw.c
index 0e15265..672a5a4 100644
--- a/drivers/media/platform/imx8/hdmi/mxc-hdmi-hw.c
+++ b/drivers/media/platform/imx8/hdmi/mxc-hdmi-hw.c
@@ -422,5 +422,11 @@ int hdmirx_startup(state_struct *state)
 	CDN_API_HDMIRX_Init_blocking(state);
 	dev_dbg(&hdmi_rx->pdev->dev,
 				"CDN_API_HDMIRX_Init_blocking() complete.\n");
+
+	/* Initialize HDMI RX CEC */
+	CDN_API_General_Write_Register_blocking(state,
+				ADDR_SINK_CAR + (SINK_CEC_CAR << 2),
+				F_SINK_CEC_SYS_CLK_EN(1) |
+				F_SINK_CEC_SYS_CLK_RSTN_EN(1));
 	return 0;
 }
-- 
1.7.9.5

